docs/sphinx/dev_guide/branch_development.rst

```diff
@@ -0,0 +1,110 @@
+.. ##
+.. ## Copyright (c) 2016-21, Lawrence Livermore National Security, LLC
+.. ## and RAJA project contributors. See the RAJA/COPYRIGHT file
+.. ## for details.
+.. ##
+.. ## SPDX-License-Identifier: (BSD-3-Clause)
+.. ##
+
+.. _branching-label:
+
+*******************************************
+RAJA Branch Development
+*******************************************
+
+========================
+Gitflow Branching Model
+========================
+
+The RAJA project uses a simple branch development model based on 
+`Gitflow <https://datasift.github.io/gitflow/IntroducingGitFlow.html>`_.
+The Gitflow model is centered around software releases. It is a simple
+workflow that makes clear which branches correspond to which phases of
+development. Those phases are represented explicitly in the branch names and
+structure of the repository. As in other branching models, developers develop 
+code on a local branch and push their work to a central repository.
+
+---------------------------------
+Persistent, Protected Branches
+---------------------------------
+
+The **main** and **develop** branches are the two primary branches we use.
+They always exist and are protected in the RAJA GitHub project in that
+changes to them only occur as a result of approved pull requests. The 
+distinction between the main and develop branches is an important part of 
+Gitflow.
+
+  * The *main* branch records the release history of the project. Each time 
+    the main branch is changed, a new tag for a new code version is made. 
+    See :ref:`semver-label` for a description of the version numbering scheme 
+    we use.
+
+  * The *develop* branch is used to integrate and test new features and most
+    bug fixes before they are merged into the main branch for a release.
+
+.. important:: **Development never occurs directly on the main branch or 
+               develop branch.**
+
+All other branches in the RAJA repo are temporary and are used to perform 
+specific development tasks. When such a branch is no longer needed, after it
+is merged for example, the branch is deleted typically.
```

Comment:
```suggestion
specific development tasks. When such a branch is no longer needed (e.g., after it
is merged), the branch is typically deleted.
```

Reply:
@kab163 thanks for proofreading and the suggested changes. I'm making modifications based on these.